native/src/routes/Pois.tsx

```diff
@@ -87,43 +94,74 @@ const Pois = ({ pois, language, cityModel, route, navigation }: PoisProps): Reac
       }
     }, RESTORE_TIMEOUT)
   }
-  const selectPoiFeature = (feature: PoiFeature | null) => {
-    if (feature && cameraRef.current) {
-      setFollowUserLocation(false)
-      navigation.setParams({ slug: feature.properties.slug })
+
+  const deselectFeature = useCallback(() => {
+    if (multipoi && currentPoi) {
+      navigation.setParams({ slug: undefined })
+    } else {
+      navigation.setParams({ slug: undefined, multipoi: undefined })
+      scrollTo(listScrollPosition)
+    }
+  }, [currentPoi, listScrollPosition, multipoi, navigation])
+  useOnBackNavigation(slug || multipoi ? deselectFeature : undefined)
+
+  const selectFeatureOnMap = (feature: PoiFeature | null) => {
+    if (!cameraRef.current) {
+      return
+    }
+    if (!feature) {
+      deselectFeature()
+      return
+    }
+    setFollowUserLocation(false)
+    if (isMultipoi(feature)) {
+      navigation.setParams({ multipoi: feature.id as number })
       scrollTo(0)
     } else {
-      navigation.setParams({ slug: undefined })
+      const poiFeature = feature.properties.pois[0]
+      navigation.setParams({ slug: poiFeature?.slug })
+      scrollTo(0)
     }
   }
 
-  const deselectPoiFeature = useCallback(() => {
-    navigation.setParams({ slug: undefined })
-    scrollTo(listScrollPosition)
-  }, [listScrollPosition, navigation])
-  useOnBackNavigation(slug ? deselectPoiFeature : undefined)
+  const selectPoiFeatureInList = useCallback(
+    (newPoiFeature: GeoJsonPoi | null) => {
+      if (!newPoiFeature || !cameraRef.current) {
+        return
+      }
+      setFollowUserLocation(false)
+      navigation.setParams({ slug: newPoiFeature.slug })
+      scrollTo(0)
+    },
+    [navigation]
+  )
 
   // Wait for followUserLocation change before moving the camera to avoid position lock
   // https://github.com/rnmapbox/maps/issues/1079
-  useEffect(() => {
-    if (!followUserLocation && selectedFeature && cameraRef.current) {
+  useLayoutEffect(() => {
+    if (!followUserLocation && currentFeatureOnMap && cameraRef.current) {
       cameraRef.current.setCamera({
-        centerCoordinate: selectedFeature.geometry.coordinates,
-        zoomLevel: selectedFeature.properties.closeToOtherPoi ? closerDetailZoom : normalDetailZoom,
+        centerCoordinate: currentFeatureOnMap.geometry.coordinates,
+        zoomLevel: closerDetailZoom,
```

Comment:
I realized that when having relatively close pois (50m), the cluster will be displayed instead of the pois. Therefore I chouse to use the closerDetailZoom. 
Maybe we'll need to think about keeping `closeToOtherPoi` to properly zoom on select. Or we just ignore it.

For now I reverted to normalDetailZoom.